core/rawdb/accessors_batch_event.go

```diff
@@ -0,0 +1,157 @@
+package rawdb
+
+import (
+	"bytes"
+	"math/big"
+	"strconv"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/ethdb"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+// ChunkRange represents the range of blocks within a chunk.
+type ChunkRange struct {
+	StartBlockNumber uint64
+	EndBlockNumber   uint64
+}
+
+// FinalizedBatchMeta holds metadata for finalized batches.
+type FinalizedBatchMeta struct {
+	BatchHash            common.Hash
+	TotalL1MessagePopped uint64
+}
+
+// WriteBatchEventSyncedL1BlockNumber updates the database with the latest synced L1 block number related to batch events in the database.
+func WriteBatchEventSyncedL1BlockNumber(db ethdb.KeyValueWriter, l1BlockNumber uint64) {
+	value := big.NewInt(0).SetUint64(l1BlockNumber).Bytes()
+	if err := db.Put(batchEventSyncedL1BlockNumberKey, value); err != nil {
+		log.Crit("failed to update synced L1 block number for batch event", "err", err)
+	}
+}
+
+// ReadBatchEventSyncedL1BlockNumber fetches the highest synced L1 block number associated with batch events from the database.
+func ReadBatchEventSyncedL1BlockNumber(db ethdb.Reader) *uint64 {
+	data, err := db.Get(batchEventSyncedL1BlockNumberKey)
+	if err != nil && isNotFoundErr(err) {
+		return nil
+	}
+	if err != nil {
+		log.Crit("failed to read synced L1 block number from database", "err", err)
+	}
+
+	number := new(big.Int).SetBytes(data)
+	if !number.IsUint64() {
+		log.Crit("unexpected synced L1 block number in database", "number", number)
+	}
+
+	batchEventSyncedL1BlockNumber := number.Uint64()
+	return &batchEventSyncedL1BlockNumber
+}
+
+// WriteBatchChunkRanges writes the block ranges for each chunk within a batch to the database.
+// It serializes the chunk ranges using RLP and stores them under a key derived from the batch index.
+func WriteBatchChunkRanges(db ethdb.KeyValueWriter, batchIndex uint64, chunkRanges []*ChunkRange) {
+	var err error
+	bytes, err := rlp.EncodeToBytes(chunkRanges)
+	if err != nil {
+		log.Crit("failed to RLP encode batch block range", "batch index", batchIndex, "err", err)
+	}
+	if err := db.Put(batchChunkRangesKey(batchIndex), bytes); err != nil {
+		log.Crit("failed to store batch block range", "batch index", batchIndex, "err", err)
+	}
+}
+
+// DeleteBatchChunkRanges removes the block ranges of all chunks associated with a specific batch from the database.
+// Note: Only committed batches can be reverted.
```

Comment:
```suggestion
// Note: Only non-finalized batches can be reverted.
```

Btw, this is not enforced in this function, is it?

Reply:
yeah. just a reminder for now.

Reply:
changed in [ba66df0](https://github.com/scroll-tech/go-ethereum/pull/515/commits/ba66df0ebcf2f7db9aa0b3203906c1f02cee623f), but still not checking has the "to be deleted" batch finalized.